src/components/fx/click.js

```diff
@@ -22,14 +22,29 @@ module.exports = function click(gd, evt, subplot) {
         hover(gd, evt, subplot, true);
     }
 
-    function emitClick() { gd.emit('plotly_click', {points: gd._hoverdata, event: evt}); }
+    function emitClick(data) { gd.emit('plotly_click', {points: data, event: evt}); }
 
-    if(gd._hoverdata && evt && evt.target) {
-        if(annotationsDone && annotationsDone.then) {
-            annotationsDone.then(emitClick);
-        } else emitClick();
+    // TODO(j): remove gd._hoverdata
+    var clickmode = gd._fullLayout.clickmode;
+    var data;
+    if(evt && evt.target) {
+        if(gd._hoverdata) {
+            data = gd._hoverdata;
+        } else if(clickmode.indexOf('anywhere') > -1) {
+            var xaxis = gd._fullLayout.xaxis;
+            var yaxis = gd._fullLayout.yaxis;
+            var bb = evt.target.getBoundingClientRect();
+            var x = xaxis.p2d(evt.clientX - bb.left);
+            var y = yaxis.p2d(evt.clientY - bb.top);
+            data = [{x: x, y: y}];
+        }
+        if(data) {
+            if(annotationsDone && annotationsDone.then) {
+                annotationsDone.then(function() { emitClick(data); });
+            } else emitClick(data);
+        }
 
         // why do we get a double event without this???
         if(evt.stopImmediatePropagation) evt.stopImmediatePropagation();
     }
-};
+};
```

Comment:
Please add a new line to the end of this file.
To catch syntax errors you could run
npm run test-syntax && npm run lint